x-pack/plugins/ml/public/application/components/data_grid/feature_importance/use_classification_path_data.tsx

```diff
@@ -165,9 +199,87 @@ export const buildClassificationDecisionPathData = ({
     }
     return undefined;
   });
+
+  // get the baseline for the current class from the trained_models metadata
+  const baselineClass = baselines.find(
+    (bl) => getStringBasedClassName(bl.class_name) === getStringBasedClassName(currentClass)
+  );
   const filteredFeatureImportance = mappedFeatureImportance.filter(
     (f) => f !== undefined
   ) as ExtendedFeatureImportance[];
 
-  return buildDecisionPathData(filteredFeatureImportance);
+  const finalResult: DecisionPathPlotData = filteredFeatureImportance
+    // sort by absolute importance so it goes from bottom (baseline) to top
+    .sort(
+      (a: ExtendedFeatureImportance, b: ExtendedFeatureImportance) =>
+        b.absImportance! - a.absImportance!
+    )
+    .map((d) => [d[FEATURE_NAME] as string, d[FEATURE_IMPORTANCE] as number, NaN]);
+
+  /**
+   * For binary classification
+   */
+  if (baselines.length === 2) {
+    // transform the numbers into the probability space
+    // starting with the baseline retrieved from trained_models metadata
+    let logOddSoFar = baselineClass?.baseline ? baselineClass?.baseline : 0;
+    for (let i = featureImportance.length - 1; i >= 0; i--) {
+      logOddSoFar += finalResult[i][1];
+      const predictionProbabilitySoFar = Math.exp(logOddSoFar) / (Math.exp(logOddSoFar) + 1);
+      finalResult[i][2] = predictionProbabilitySoFar;
+    }
+    return finalResult;
+  }
```

Comment:
worth extracting to the dedicated function

Reply:
Updated here 10eae2d